src/Components/reducers/videoReducer.jsx

```diff
@@ -0,0 +1,58 @@
+export const videoReducer = (state, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case "MUSIC":
+      return {
+        ...state,
+        categories: {
+          ...state.categories,
+          Music: !state.categories.Music,
+        },
+      };
+    case "TV_SHOWS":
+      return {
+        ...state,
+        categories: {
+          ...state.categories,
+          TV_Shows: !state.categories.TV_Shows,
+        },
+      };
+    case "LEARNING":
+      return {
+        ...state,
+        categories: {
+          ...state.categories,
+          Learning: !state.categories.Learning,
+        },
+      };
```

Comment:
you should not make different case for every filter, you can optimize this by using only one case for all filters